config/Version.public.xcconfig

```diff
@@ -1,2 +1,2 @@
-VERSION_LONG = 23.7.0.20231115
+VERSION_LONG = 23.7.0.20231116
```

Comment:
This should have been `23.7.0.3`, but there was a bug in the beta lane which was not noticed in 23.6 because we didn't ship any new beta and I missed at the time of the first and only beta, see https://github.com/wordpress-mobile/WordPress-iOS/pull/22049/files#diff-10b48950e900a8410c64f50f103cc4a3c04cea95a38c6aab84de7db0a5c1e75f

https://github.com/wordpress-mobile/WordPress-iOS/pull/22087 fixes the issue that resulted in this incorrect value. I didn’t think it worth going through the release process again, because I would have had to hack the automation to skip updating the GitHub label, as that would have failed since the previous release build already did it. Also, I have a feeling that Apple would have rejected a build with 23.7.0.3 when the previous one was 23.7.0.20231116 as 3 is less than 20231116.